src/submission/infrastructure/types.ts

```diff
@@ -0,0 +1,20 @@
+import { uuidType } from 'typesafe-uuid';
+
+export class SubmissionId extends uuidType<'SubmissionId'>() {}
+
+export interface SubmissionRepository {
+    findAll(): Promise<SubmissionDTO[]>;
+    findById(id: SubmissionId): Promise<SubmissionDTO | null>;
+    save(sub: SubmissionDTO): Promise<SubmissionDTO | null>;
```

Comment:
should `save` be replaced with `create` and `update` (following the CRUD pattern)